src/Engine/ProtoCore/Core.cs

```diff
@@ -762,6 +762,28 @@ public void GenerateExecutable()
             DSExecutable.CurrentDSFileName = CurrentDSFileName;           
         }
 
+        /// <summary>
+        /// Gets the size to be used for runtime tables of symbols, procedures and instruction streams.
+        /// Note: since blocks are stored consecutively but may have gaps due to procedures being deleted,
+        /// this is based on largest id rather than amount of blocks.
+        /// </summary>
+        internal int GetRuntimeTableSize()
+        {
+            // Due to the way this list is constructed, the largest id is the one of the last block.
+            var lastBlock = CompleteCodeBlockList.LastOrDefault();
+            if (lastBlock == null)
+            {
+                return 0;
+            }
```

Comment:
Could you explain this change?

Reply:
Sure. I'll add a comment in code.

Reply:
I meant for reviewers to understand the change.

Reply:
Sorry, I thought if it was added to the code it would be useful for future reference too. I'll copy it here as well:
```
            // If there are no code blocks yet, then the required size for tables is 0.
            // This happens when the first code block is being created and its id is being generated.
```